Sources/StreamChat/ChatClient_Tests.swift

```diff
@@ -617,6 +612,213 @@ class ChatClient_Tests: StressTestCase {
             XCTAssertEqual(clientInitCompletionError as? TestError, error)
         }
     }
+    
+    func test_reloadUserIfNeededIsNotCalled_whenClientIsInitialized_andTokenProviderIsNil() {
+        _ = ChatClient(
+            config: inMemoryStorageConfig,
+            workerBuilders: [],
+            eventWorkerBuilders: [],
+            environment: testEnv.environment
+        )
+        
+        // `reloadUserIfNeeded` is not called on `clientUpdater` hence it wasn't lazily initialised
+        XCTAssertNil(testEnv.clientUpdater)
```

Comment:
this is just waiting to get randomly broken in the future :) Can we do something like `XCTAssert(testEnv.clientUpdater?.reloadUserIfNeeded_called != true)` instead?